app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -1,11 +1,22 @@
 package com.example.sharingang
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
 import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import com.example.sharingang.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        val binding: ActivityMainBinding =
+            DataBindingUtil.setContentView(this, R.layout.activity_main)
+       binding.description = intent.getStringExtra("LAST_DESCRIPTION")
```

Comment:
How can you access intent without declaring it first?

Reply:
So there's a [getIntent](https://developer.android.com/reference/android/app/Activity.html#getIntent()) method on Activity, which you have access to since you're inheriting from that class (indirectly).

In Kotlin, it's customary to use [properties](https://kotlinlang.org/docs/properties.html) instead.

The final bit of magic, is that Android's APIs, through the Kotlin shims wrapping them, have properties inserted instead of `getX` `setX` pairs.

So, instead of calling `this.getIntent()`, you can instead just write `this.intent`, or even `intent` as a shorthand.

Reply:
Perfect explanation, thanks!